packages/@blockprotocol/graph/src/stdlib/subgraph/edge/link-entity.ts

```diff
@@ -0,0 +1,225 @@
+import { Entity, EntityId } from "../../../types/entity.js";
+import { OutwardEdge, Subgraph } from "../../../types/subgraph.js";
+import {
+  isHasLeftEntityEdge,
+  isHasRightEntityEdge,
+  isIncomingLinkEdge,
+  isOutgoingLinkEdge,
+  OutgoingLinkEdge,
+} from "../../../types/subgraph/edges/outward-edge-alias.js";
+import { mustBeDefined } from "../../must-be-defined.js";
+import { getEntity } from "../element/entity.js";
+
+const convertTimeToStringWithDefault = (
+  timestamp?: Date | string,
+  defaultConstructor = () => new Date().toISOString(),
+) => {
+  return timestamp === undefined
+    ? defaultConstructor()
+    : typeof timestamp === "string"
+    ? timestamp
+    : timestamp.toISOString();
+};
+
+const getUniqueEntitiesFilter = () => {
+  const set = new Set();
+  return (entity: Entity) => {
+    const editionIdString = JSON.stringify(entity.metadata.editionId);
+    if (set.has(editionIdString)) {
+      return false;
+    } else {
+      set.add(editionIdString);
+      return true;
+    }
+  };
+};
+
+/** @todo - Update these methods to take intervals instead of timestamps */
+
+/**
+ * For a given moment in time, get all outgoing link entities from a given entity.
+ *
+ * @param subgraph
+ * @param {EntityId} entityId - The ID of the source entity to search for outgoing links from
+ * @param {Date | string} [timestamp] - An optional `Date` or an ISO-formatted datetime string of the moment to search
+ *    for, if not supplied it defaults to the current time
+ */
+export const getOutgoingLinksForEntity = (
+  subgraph: Subgraph,
+  entityId: EntityId,
+  timestamp?: Date | string,
+): Entity[] => {
+  const timestampString = convertTimeToStringWithDefault(timestamp);
+
+  const entityEdges = subgraph.edges[entityId];
+
+  if (!entityEdges) {
+    return [];
+  }
+
+  const uniqueEntitiesFilter = getUniqueEntitiesFilter();
+
+  return (
+    Object.entries(entityEdges)
+      // Only look at outgoing edges that were created before or at the timestamp
+      .filter(([edgeTimestamp, _]) => edgeTimestamp <= timestampString)
```

Comment:
Is there a reason you prefer this? I like it being explicit that we're destructuring the tuple here

Reply:
Just seems a bit pointless to me – destructuring in JS tends to be 'take what you need, and no more'.

The fact that we're destructuring a tuple should be clear from the value implied by `Object.entries` and the syntax. If people don't know what's produced by that, then I'm not sure introducing a mysterious `_` is going to help.

It's not a big deal to leave it in but it'll probably look weird to the next JS developer who'll remove it (which also won't be a big deal, so let's not sweat it)

Reply:
I would agree if it wasn't a `filter`. As in if this was a `map` or something and you were destructuring to just select one part of it, that'd make more sense to me, because you don't need to care about the next thing, but because the second element is genuinely used in the next function, I'd prefer it to be explicit, otherwise it takes that extra second to understand what's going on

Reply:
Fair enough. If it's for clarity I would consider using variable names like `_edge` because it makes it even easier to see what's going on, but isn't important to do in this PR

Reply:
Tiny change to do so I'll driveby and then merge. Cheers